File: 0028-find-the-index-of-the-first-occurrence-in-a-string/0028-find-the-index-of-the-first-occurrence-in-a-string.cpp
class Solution {
public:
    int strStr(string haystack, string needle) {
        int n = haystack.size(),k = needle.size();
        //auto ptr = haystack.begin();
        if(k>n){return -1;}
        for(int i=0;i<=n-k;i++){
            if(needle == haystack.substr(i,k)){return i;}
        }
        return -1;
    }
};

File: 0028-find-the-index-of-the-first-occurrence-in-a-string/0028-find-the-index-of-the-first-occurrence-in-a-string.py
class Solution:
    def strStr(self, haystack: str, needle: str) -> int:
        big=len(haystack)
        small=len(needle)
        if small>big:
            return -1
        for i in range(0,big+1-small):
            if needle[0]==haystack[i]:
                if needle==haystack[i:i+small]:
                    return i
        return -1